Add default value for array properties

VincentLanglet opened the pull request:
## Subject

I am targeting this branch, because BC.

Multiple values are expected to be a not nullable array but are not explicitly set to `[]` by default.
This gave me error when I tried to use the master branch since getter was expecting to return an array.

## Changelog

```markdown
### Fixed
- Fixed default `[]` value for every non-nullable array class properties
```

jordisala1991 commented:
This default values change could cause BC breaks if you rely on the previous default. I think I remember other PRs were the default value was not changed but we left a comment for next major, do you remember?

VincentLanglet commented:
> This default values change could cause BC breaks if you rely on the previous default. I think I remember other PRs were the default value was not changed but we left a comment for next major, do you remember?

It was array with a default `false` value `AbstractAdmin`.
But they were explicitly typed as `array|false`.

The only default value I found using the fact it was a `null` value was the `result` from the `Datagrid` and I changed the phpdoc to `array|null`. But for other I didn't find `null` checks and getters were expecting to return an array. If you add typehint it'll cause error. In other words, we could say that it was a bugfix since I fixed the code to respect the contract of the phpdoc.

phansys commented:
>This default values change could cause BC breaks if you rely on the previous default. I think I remember other PRs were the default value was not changed but we left a comment for next major, do you remember?

I think you mean cases like this: https://github.com/sonata-project/SonataAdminBundle/commit/ec6eec20300a6ea509dd22824774333ef89544d3#diff-61e6d1f8fafe87b868adbe28ce0ba8f6R517-R518.

VincentLanglet commented:
Is that ok for you @jordisala1991 ? I saw you 👍  my comment, but you didn't approve the PR.
